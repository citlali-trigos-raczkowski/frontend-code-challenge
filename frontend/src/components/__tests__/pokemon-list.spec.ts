import { PokemonGalleryItemMockProp } from "../../../tests/mocks/mock-props";
import { shallowMount, createLocalVue } from "@vue/test-utils";
import PokemonList from "../pokemon-list.vue";
import VueRouter from "vue-router";

const localVue = createLocalVue();
localVue.use(VueRouter);

describe("components/pokemon-list.vue", () => {
  it("renders correctly", () => {
    const wrapper = shallowMount(PokemonList, {
      propsData: PokemonGalleryItemMockProp,
      localVue,
    });
    expect(wrapper.element).toMatchSnapshot();
  });

  it("includes link to the pokemon", async () => {
    const wrapper = shallowMount(PokemonList, {
      propsData: PokemonGalleryItemMockProp,
      localVue,
    });

    const aWrapper = wrapper.find("router-link-stub");
    expect(aWrapper.attributes().to).toBe(
      PokemonGalleryItemMockProp.pokemon.name
    );
  });

  it("heart toggle listens", async () => {
    const wrapper = shallowMount(PokemonList, {
      propsData: PokemonGalleryItemMockProp,
      localVue,
    });
    const heartToggle = wrapper.find("heart-toggle-stub");
    heartToggle.trigger("togglepokemonfavorite");

    expect(heartToggle.attributes().heartcolor).toBe("pink");
  });
});
