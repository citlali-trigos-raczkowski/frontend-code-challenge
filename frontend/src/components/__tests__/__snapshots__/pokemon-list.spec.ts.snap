// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/pokemon-list.vue renders correctly 1`] = `
<v-list-item-stub
  activeclass=""
  data-cy="pokemon-list-item"
  tag="div"
>
  <router-link-stub
    ariacurrentvalue="page"
    event="click"
    id="pokemon-list-link"
    tag="a"
    to="Ivysaur"
  >
    <v-list-item-avatar-stub
      size="40"
    >
      <v-img-stub
        options="[object Object]"
        position="center center"
        src="https://img.pokemondb.net/artwork/ivysaur.jpg"
        transition="fade-transition"
      />
    </v-list-item-avatar-stub>
     
    <v-list-item-content-stub>
      <v-list-item-title-stub>
        Ivysaur
      </v-list-item-title-stub>
    </v-list-item-content-stub>
  </router-link-stub>
   
  <heart-toggle-stub
    heartcolor="pink"
    id="pokemon-list-heart"
    togglepokemonfavorite="[Function]"
  />
</v-list-item-stub>
`;
